commands/databases.go

```diff
@@ -167,11 +170,35 @@ func RunDatabaseCreate(c *CmdConfig) error {
 		return err
 	}
 
-	db, err := c.Databases().Create(r)
+	dbs := c.Databases()
+
+	db, err := dbs.Create(r)
+	if err != nil {
+		return err
+	}
+
+	wait, err := c.Doit.GetBool(c.NS, doctl.ArgCommandWait)
 	if err != nil {
 		return err
 	}
 
+	if wait {
+		dbs := c.Databases()
+		notice("Database creation is in progress, waiting for database to be online")
+
+		err := waitForDatabaseReady(dbs, db.ID)
+		if err != nil {
+			return fmt.Errorf(
+				"database couldn't enter the `online` state: %v",
+				err,
+			)
+		}
+
+		db, _ = dbs.Get(db.ID)
```

Comment:
```suggestion
		connection := db.Connection
		dbs := c.Databases()
		notice("Database creation is in progress, waiting for database to be online")

		err := waitForDatabaseReady(dbs, db.ID)
		if err != nil {
			return fmt.Errorf(
				"database couldn't enter the `online` state: %v",
				err,
			)
		}

		db, _ = dbs.Get(db.ID)
		db.Connection = connection
```

As mentioned in https://github.com/digitalocean/doctl/issues/1263,  some database engines (`mongodb` now, but potentially others) only return the password in the initial response to the create request. It is not included in subsequent GET responses. So that value must be retained from the initial response so that the connection URI contains the password.

For example, without this change, the output of a MongoDB create is:

```
$ ./builds/doctl databases create test-01 --engine mongodb --version 4 --wait
Notice: Database creation is in progress, waiting for database to be online
.....................................
Notice: Database created
ID                                      Name       Engine     Version    Number of Nodes    Region    Status    Size              URI                                                                                                                    Created At
6da96d00-9e06-4d4c-87e0-c6a6642e3cc1    test-01    mongodb    4          1                  nyc1      online    db-s-1vcpu-1gb    mongodb+srv://doadmin:@test-01-3ff22e43.mongo.ondigitalocean.com/admin?tls=true&authSource=admin&replicaSet=test-01    2022-10-14 16:41:33 +0000 UTC
```
Notice the missing password in the connection URI (`doadmin:@` instead of `doadmin:<password>@`).